exchange/tx.go

```diff
@@ -269,26 +270,46 @@ func (tx *Tx) buildRoot() error {
 
 // Ref returns the DataRef associated with this transaction
 func (tx *Tx) Ref() *DataRef {
-	return &DataRef{
+	ref := &DataRef{
 		PayloadCID:  tx.root,
 		StoreID:     tx.storeID,
 		PayloadSize: tx.size,
 	}
+
+	if len(tx.entries) <= 0 {
```

Comment:
`len` can't be negative

Reply:
I know but ```len``` returns an int, what if there is a cosmic ray that flips the bit that signs for a + into a - ? 😁